osu.Game/Rulesets/Edit/IPositionSnapProvider.cs

```diff
@@ -5,23 +5,27 @@
 
 namespace osu.Game.Rulesets.Edit
 {
+    /// <summary>
+    /// A snap provider which given a proposed position for a hit object, potentially offers a more correct position and time value inferred from the context of the beatmap.
+    /// Provided values are inferred in an isolated context, without consideration of other nearby hit objects.
+    /// </summary>
     public interface IPositionSnapProvider
     {
         /// <summary>
         /// Given a position, find a valid time and position snap.
         /// </summary>
         /// <remarks>
-        /// This call should be equivalent to running <see cref="SnapScreenSpacePositionToValidPosition"/> with any additional logic that can be performed without the time immutability restriction.
+        /// This call should be equivalent to running <see cref="FindSnappedPosition"/> with any additional logic that can be performed without the time immutability restriction.
         /// </remarks>
         /// <param name="screenSpacePosition">The screen-space position to be snapped.</param>
         /// <returns>The time and position post-snapping.</returns>
-        SnapResult SnapScreenSpacePositionToValidTime(Vector2 screenSpacePosition);
+        SnapResult FindSnappedPositionAndTime(Vector2 screenSpacePosition);
 
         /// <summary>
         /// Given a position, find a value position snap, restricting time to its input value.
```

Comment:
I understand this part existed before this PR, but since we're cleaning things here...
```suggestion
        /// Given a position, find a valid position snap, restricting time to its input value.
```

Also I'm not sure what the point of this `SnapScreenSpacePositionToValidPosition`/`FindSnappedPosition` method is and why it returns `SnapResult` in the first place, feels like it could indeed be simplified to a boolean (or just not exist in the first place and not pay attention to the resultant time value..?)

At least it makes much more sense now with the rename.

Reply:
Let's keep return value changes out of this pull please. Maybe separately if it's deemed better.

Reply:
Definitely for a separate PR, just bringing it here for note.